docs/user/metal/install_ipi.md

```diff
@@ -0,0 +1,264 @@
+      internal DNS requirements.
+  * ***External DNS*** - While the cluster automates the internal DNS
+    requirements, two external DNS records must be created in whatever DNS
+    server is appropriate for this environment.
+    * `api.<cluster-name>.<base-domain>` - pointing to the API VIP
+    * `*.apps.<cluster-name>.<base-domain>` - pointing to the Ingress VIP
+
+* **NIC #2 - Provisioning Network**
+  * A private, non-routed network, used for PXE based provisioning.
+  * DHCP is automated for this network.
+  * Addressing for this network is currently hard coded as `172.22.0.0/24`, but
+    will be made configurable in the future.
+
+* **Out-of-band Management Network**
+  * Servers will typically have an additional NIC used by the onboard
+    management controllers (BMCs).  These BMCs must be accessible and routed to
+    the host.
+
+### Provisioning Host
+
+The installer must be run from a host that is attached to the same networks as
+the cluster, as described in the previous section.  We refer to this host as
+the *provisioning host*.  The easiest way to provide a provisioning host is to
+use one of the hosts that is intended to later become a worker node in the same
+cluster.  That way it is already connected to the proper networks.
+
+It is recommended that the provisioning host be a bare metal host, as it must be
+able to use libvirt to launch the OpenShift bootstrap VM locally.
+
+### Supported Hardware
+
+The architecture is intended to support a wide variety of hardware.  This was
+one of the reasons Ironic is used as an underlying technology.  However, so far
+development and testing has focused on PXE based provisioning using IPMI for
+out-of-band management of hosts.  Other provisioning approaches will be added,
+tested, and documented over time.
+
+## Installation Process
+
+Once an environment has been prepared according to the documented
+pre-requisites, the install process is the same as other IPI based platforms.
+
+`openshift-install create cluster`
+
+However, it is recommended to prepare an `install-config.yaml` file in advance,
+containing all of the details of the bare metal hosts to be provisioned.
+
+### Install Config
+
+The `install-config.yaml` file requires some additional details.  Most of the
+information is teaching the installer and the resulting cluster enough about
+the available hardware so that it is able to fully manage it.
+
+Here is an example `install-config.yaml` with the required `baremetal` platform
+details.
+
+**IMPORTANT NOTE:** The current install configuration for the `baremetal`
+platform should be considered experimental and still subject to change without
+backwards compatibility.  In particular, some items likely to change soon
+include:
+
+* The `image` section will get completely removed.
+
+```yaml
+apiVersion: v1beta4
+baseDomain: test.metalkube.org
+metadata:
+  name: ostest
+compute:
+- name: worker
+  replicas: 1
```

Comment:
You've only listed 3 hosts below ... do you need a 4th for this worker?

Reply:
Workers not integrated into installer yet, it's done later by dev-scripts

Reply:
@markmc good point, I'll add a 4th host to represent a worker

@stbenjam I've glossed over that fact for the moment.  Maybe we should have a section at the end though that covers various pending PRs, and the workarounds in the meantime

Reply:
Note: A 4th host in the install-config won't be created in Ironic, it'll just be ignored.  That's something that needs to get fixed.  But at the moment it's the wrong Ironic anyway, as the data isn't transferred from bootstrap Ironic to the running cluster's Ironic.

Reply:
I've added a new section that describes key bits of integration work that are still WIP

Reply:
@stbenjam and yeah, I know it's not handled yet, but I think it's correct in terms of documenting what the `install-config.yaml` should look like.  Once the installer creates the `BareMetalHosts`, worker details need to be in here.